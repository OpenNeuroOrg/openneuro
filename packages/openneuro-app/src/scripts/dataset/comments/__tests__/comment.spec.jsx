import React from "react"
import { render } from "@testing-library/react"
import Comment from "../comment.jsx"
import formatDistanceToNow from "date-fns/formatDistanceToNow"

vi.mock("date-fns/formatDistanceToNow")
vi.mock("../../../config.ts")

const emptyState =
  '{"blocks":[{"key":"3sm42","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}'

describe("Comment component", () => {
  it("renders with an empty comment", () => {
    formatDistanceToNow.mockReturnValueOnce("almost 2 years")

    const wrapper = render(
      <Comment
        data={{
          id: "9001",
          text: emptyState,
          user: {
            id: "1234",
            email: "example@example.com",
            name: "Example Exampler",
          },
          createDate: new Date("2019-04-02T19:56:41.222Z").toISOString(),
        }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it("renders an ORCID user comment", () => {
    formatDistanceToNow.mockReturnValueOnce("almost 2 years")

    const wrapper = render(
      <Comment
        data={{
          id: "9001",
          text: emptyState,
          user: {
            id: "1234",
            email: "example@example.com",
            name: "Example Exampler",
            orcid: "1234-5678-9101",
          },
          createDate: new Date("2019-04-02T19:56:41.222Z").toISOString(),
        }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
