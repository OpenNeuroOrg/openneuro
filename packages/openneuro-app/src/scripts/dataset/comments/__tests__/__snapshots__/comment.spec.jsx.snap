// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Comment component > renders an ORCID user comment 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="comment"
      >
        <div
          class="row comment-header"
        >
          By 
          <a
            href="/user/1234-5678-9101"
          >
            Example Exampler
          </a>
           
          <a
            href="https://sandbox.orcid.org/1234-5678-9101"
          >
            <img
              alt="ORCID logo"
              height="16"
              src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='utf-8'?%3e%3c!--%20Generator:%20Adobe%20Illustrator%2019.1.0,%20SVG%20Export%20Plug-In%20.%20SVG%20Version:%206.00%20Build%200)%20--%3e%3csvg%20version='1.1'%20id='Layer_1'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%20x='0px'%20y='0px'%20viewBox='0%200%20256%20256'%20style='enable-background:new%200%200%20256%20256;'%20xml:space='preserve'%3e%3cstyle%20type='text/css'%3e%20.st0{fill:%23A6CE39;}%20.st1{fill:%23FFFFFF;}%20%3c/style%3e%3cpath%20class='st0'%20d='M256,128c0,70.7-57.3,128-128,128C57.3,256,0,198.7,0,128C0,57.3,57.3,0,128,0C198.7,0,256,57.3,256,128z'/%3e%3cg%3e%3cpath%20class='st1'%20d='M86.3,186.2H70.9V79.1h15.4v48.4V186.2z'/%3e%3cpath%20class='st1'%20d='M108.9,79.1h41.6c39.6,0,57,28.3,57,53.6c0,27.5-21.5,53.6-56.8,53.6h-41.8V79.1z%20M124.3,172.4h24.5%20c34.9,0,42.9-26.5,42.9-39.7c0-21.5-13.7-39.7-43.7-39.7h-23.7V172.4z'/%3e%3cpath%20class='st1'%20d='M88.7,56.8c0,5.5-4.5,10.1-10.1,10.1c-5.6,0-10.1-4.6-10.1-10.1c0-5.6,4.5-10.1,10.1-10.1%20C84.2,46.7,88.7,51.3,88.7,56.8z'/%3e%3c/g%3e%3c/svg%3e"
              width="16"
            />
          </a>
           - almost 2 years ago
        </div>
        <div
          class="row comment-body"
        >
          <div
            class="DraftEditor-root"
          >
            <div
              class="DraftEditor-editorContainer"
            >
              <div
                class="public-DraftEditor-content"
                contenteditable="false"
                spellcheck="false"
                style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
              >
                <div
                  data-contents="true"
                >
                  <div
                    class=""
                    data-block="true"
                    data-editor="9001"
                    data-offset-key="3sm42-0-0"
                  >
                    <div
                      class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                      data-offset-key="3sm42-0-0"
                    >
                      <span
                        data-offset-key="3sm42-0-0"
                      >
                        <br
                          data-text="true"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row replies"
      >
        <div
          class="comment-reply"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="comment"
    >
      <div
        class="row comment-header"
      >
        By 
        <a
          href="/user/1234-5678-9101"
        >
          Example Exampler
        </a>
         
        <a
          href="https://sandbox.orcid.org/1234-5678-9101"
        >
          <img
            alt="ORCID logo"
            height="16"
            src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='utf-8'?%3e%3c!--%20Generator:%20Adobe%20Illustrator%2019.1.0,%20SVG%20Export%20Plug-In%20.%20SVG%20Version:%206.00%20Build%200)%20--%3e%3csvg%20version='1.1'%20id='Layer_1'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%20x='0px'%20y='0px'%20viewBox='0%200%20256%20256'%20style='enable-background:new%200%200%20256%20256;'%20xml:space='preserve'%3e%3cstyle%20type='text/css'%3e%20.st0{fill:%23A6CE39;}%20.st1{fill:%23FFFFFF;}%20%3c/style%3e%3cpath%20class='st0'%20d='M256,128c0,70.7-57.3,128-128,128C57.3,256,0,198.7,0,128C0,57.3,57.3,0,128,0C198.7,0,256,57.3,256,128z'/%3e%3cg%3e%3cpath%20class='st1'%20d='M86.3,186.2H70.9V79.1h15.4v48.4V186.2z'/%3e%3cpath%20class='st1'%20d='M108.9,79.1h41.6c39.6,0,57,28.3,57,53.6c0,27.5-21.5,53.6-56.8,53.6h-41.8V79.1z%20M124.3,172.4h24.5%20c34.9,0,42.9-26.5,42.9-39.7c0-21.5-13.7-39.7-43.7-39.7h-23.7V172.4z'/%3e%3cpath%20class='st1'%20d='M88.7,56.8c0,5.5-4.5,10.1-10.1,10.1c-5.6,0-10.1-4.6-10.1-10.1c0-5.6,4.5-10.1,10.1-10.1%20C84.2,46.7,88.7,51.3,88.7,56.8z'/%3e%3c/g%3e%3c/svg%3e"
            width="16"
          />
        </a>
         - almost 2 years ago
      </div>
      <div
        class="row comment-body"
      >
        <div
          class="DraftEditor-root"
        >
          <div
            class="DraftEditor-editorContainer"
          >
            <div
              class="public-DraftEditor-content"
              contenteditable="false"
              spellcheck="false"
              style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
            >
              <div
                data-contents="true"
              >
                <div
                  class=""
                  data-block="true"
                  data-editor="9001"
                  data-offset-key="3sm42-0-0"
                >
                  <div
                    class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                    data-offset-key="3sm42-0-0"
                  >
                    <span
                      data-offset-key="3sm42-0-0"
                    >
                      <br
                        data-text="true"
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row replies"
    >
      <div
        class="comment-reply"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Comment component > renders with an empty comment 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="comment"
      >
        <div
          class="row comment-header"
        >
          By 
          Example Exampler
           - almost 2 years ago
        </div>
        <div
          class="row comment-body"
        >
          <div
            class="DraftEditor-root"
          >
            <div
              class="DraftEditor-editorContainer"
            >
              <div
                class="public-DraftEditor-content"
                contenteditable="false"
                spellcheck="false"
                style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
              >
                <div
                  data-contents="true"
                >
                  <div
                    class=""
                    data-block="true"
                    data-editor="9001"
                    data-offset-key="3sm42-0-0"
                  >
                    <div
                      class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                      data-offset-key="3sm42-0-0"
                    >
                      <span
                        data-offset-key="3sm42-0-0"
                      >
                        <br
                          data-text="true"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row replies"
      >
        <div
          class="comment-reply"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="comment"
    >
      <div
        class="row comment-header"
      >
        By 
        Example Exampler
         - almost 2 years ago
      </div>
      <div
        class="row comment-body"
      >
        <div
          class="DraftEditor-root"
        >
          <div
            class="DraftEditor-editorContainer"
          >
            <div
              class="public-DraftEditor-content"
              contenteditable="false"
              spellcheck="false"
              style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
            >
              <div
                data-contents="true"
              >
                <div
                  class=""
                  data-block="true"
                  data-editor="9001"
                  data-offset-key="3sm42-0-0"
                >
                  <div
                    class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                    data-offset-key="3sm42-0-0"
                  >
                    <span
                      data-offset-key="3sm42-0-0"
                    >
                      <br
                        data-text="true"
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row replies"
    >
      <div
        class="comment-reply"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
